### PR TITLE
ec2_group multi account

### PR DESCRIPTION
**Pull request by ryansb**
##### SUMMARY

Another fix for #44788

##### ISSUE TYPE

- Bugfix Pull Request

##### ANSIBLE VERSION

```paste below

```

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by s-hertel — APPROVED** [review]
Looks good. Shippable failures are related, as the tests are unstable, but not related to this bugfix/feature. Need to run the tests until they pass.

File: lib/ansible/modules/cloud/amazon/ec2_group.py [hunk 1]
```diff
@@ -492,14 +492,15 @@ def get_target_from_rule(module, client, rule, name, group, groups, vpc_id):
     target_group_created = False
 
     validate_rule(module, rule)
-    if rule.get('group_id') and re.match(FOREIGN_SECURITY_GROUP_REGEX, rule['group_id']) and current_account_id not in rule['group_id']:
+    if rule.get('group_id') and re.match(FOREIGN_SECURITY_GROUP_REGEX, rule['group_id']):
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/ec2_group.py:495]: Nervous about backporting as this adds new functionality.

**Comment by mattclay** [comment]
It looks like we have a permissions issue for the tests: 

> An error occurred (UnauthorizedOperation) when calling the DescribeVpcPeeringConnections operation: You are not authorized to perform this operation.

File: lib/ansible/modules/cloud/amazon/ec2_group.py (hunk 1 above)
Comment (s-hertel) [reply lib/ansible/modules/cloud/amazon/ec2_group.py:495]: Oh, testing 2.5, it looks like this was supported at the time as well. There's just a lack of documentation regarding it. Not new at all.

File: test/integration/targets/ec2_group/tasks/main.yml [hunk 2]
```diff
@@ -9,7 +9,13 @@
 # - include: ../../setup_ec2/tasks/common.yml module_name: ec2_group
 
 - include: ./credential_tests.yml
-- block:
+- module_defaults:
```
Comment (s-hertel) [inline test/integration/targets/ec2_group/tasks/main.yml:12]: Will module_defaults be backported? Or should these tests be updated to use module_defaults after the cp has been done?

Comment (ryansb) [reply test/integration/targets/ec2_group/tasks/main.yml:12]: module_defaults was in 2.7, so for backporting to 2.7 this is fine. For 2.6 we'll have to either change to not use module_defaults or not backport the integration test at all.

**Pull request merged**